pkg/operator/ceph/controller/spec.go

```diff
@@ -653,8 +655,27 @@ func PrivilegedContext(runAsRoot bool) *v1.SecurityContext {
 	return sec
 }
 
-// LogCollectorContainer runs a cron job to rotate logs
+// LogCollectorContainer rotate logs
 func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Container {
+
+	maxLogSize := "500M"
+	if c.LogCollector.MaxLogSize != nil {
+		maxLogSize = fmt.Sprintf("%dM", c.LogCollector.MaxLogSize.Value()/1000/1000)
+		if maxLogSize == "0M" {
+			logger.Info("maxLogSize is 0M setting to minimum of 1M")
+			maxLogSize = "1M"
+		}
+	}
+
+	periodicity := "daily"
+	if c.LogCollector.Periodicity == "1h" {
+		periodicity = "hourly"
+	} else if c.LogCollector.Periodicity == "24h" || c.LogCollector.Periodicity == "1d" {
+		periodicity = "daily"
+	} else if c.LogCollector.Periodicity != "" {
+		periodicity = c.LogCollector.Periodicity
```

Comment:
> The following would verify:
> 
> * Create a Rook cluster without your changes (e.g. master cluster is fine). Set the periodicity to `2d` (or something else that we don't support anymore)
> * Upgrade to your changes:
>   
>   1. Upgrade to the new schema: `kubectl apply -f crds.yaml`.
>   2. Run the new operator with your changes
>   3. Validate if `2h` is still allowed in the schema or if it is rejected.

with the same `2h` it allowed but when `3h` it rejected which is kind of expected since there was no change in schema with `2h` no action applied but with 3h there was change in schema and it wasrejected
> 
> I suspect `kubectl get cephcluster -o yaml` will still show `2h` and the new script will also have `2h` formatted instead of `daily` as we need.

yes, right that's why I added a new check in if, pointed below so that we get periodicity in the desired format.

Reply:
user will give in that format only but it is the cr validation that will be using regex.